packages/runtime-common/loader.ts

```diff
@@ -435,10 +435,9 @@ export class Loader {
     urlOrRequest: string | URL | Request,
     init?: RequestInit,
   ): Promise<Response> {
+    let request = this.asRequest(urlOrRequest, init);
```

Comment:
i needed to make this adjustment so that we could actually mutate the request in the loader url handlers (this let's us add a special header to the request). Otherwise any mutation you do to the request is just thrown away.